pkg/logql/quantile_over_time_sketch.go

```diff
@@ -283,6 +283,11 @@ func JoinQuantileSketchVector(next bool, r StepResult, stepEvaluator StepEvaluat
 		if stepEvaluator.Error() != nil {
 			return nil, stepEvaluator.Error()
 		}
+		// Don't continue to evaluate more steps if we already filled the
+		// # of steps we should have based on start/end and step params.
+		if len(result) == stepCount {
```

Comment:
How's that done in the original join?

Reply:
Found it https://github.com/grafana/loki/blob/918e7365dbbe4ac33ec02b803c6cf7155eb4a9e1/pkg/logql/engine.go#L388